distribution/test/itests/test-itests-common/src/test/java/ddf/common/test/mock/csw/FederatedCswMockServer.java

```diff
@@ -0,0 +1,207 @@
+     * Provides itest verification when a specific Http call is made.
+     *
+     * @return VerifyHttp using verifyHttp method with the tub server as the parameter.
+     */
+    public VerifyHttp verifyHttp() {
+        return VerifyHttp.verifyHttp(cswStubServer);
+    }
+
+    /**
+     * Get the port being used by the stub server.
+     *
+     * @return the port being used by the stub server.
+     */
+    public int getPort() {
+        return port;
+    }
+
+    /**
+     * Get the root url being used by the stub server.
+     *
+     * @return the root url being used by the stub server.
+     */
+    public String getRoot() {
+        return httpRoot;
+    }
+
+    /**
+     * Returns the restito stub server to allow for any unsupported actions
+     *
+     * @return the stub server managed by this class
+     */
+    public StubServer getServer() {
+        return cswStubServer;
+    }
+
+    /**
+     * Set a specific capability Response. Overwrites default/previous response condition.
+     *
+     * @param resourceName name of capability file
+     * @throws IOException
+     */
+    public void setCapabilityResponse(String resourceName) throws IOException {
+        try (InputStream inputStream = getClass().getResourceAsStream("/" + resourceName)) {
+            String document = substituteTags(IOUtils.toString(inputStream));
+
+            LOGGER.debug("Capability response: \n{}", document);
+
+            whenHttp().match(get("/services/csw"))
+                    .then(ok(), contentType("text/xml"), bytesContent(document.getBytes()));
+        }
+    }
+
+    /**
+     * Set a specific query Response. Overwrites default/previous response condition.
+     *
+     * @param resourceName name of resource file
+     * @throws IOException
+     */
+    public void setQueryResponse(String resourceName) throws IOException {
+        try (InputStream inputStream = getClass().getResourceAsStream("/" + resourceName)) {
+            String document = substituteTags(IOUtils.toString(inputStream));
+
+            LOGGER.debug("Query response: \n{}", document);
+
+            whenHttp().match(post("/services/csw"), withPostBodyContaining("GetRecords"))
+                    .then(ok(), contentType("text/xml"), bytesContent(document.getBytes()));
+        }
+    }
+
+    private void setDefaultCapabilityResponse() throws IOException {
+        setCapabilityResponse("csw-get-capabilities-response.xml");
+    }
+
+    private void setDefaultQueryResponse() throws IOException {
+        setQueryResponse("csw-query-response.xml");
+    }
+
+    private String substituteTags(String document) {
+        String resultDocument = substituteSourceId(document);
+        resultDocument = substitutePortNumber(resultDocument);
+        return substituteHttpRoot(resultDocument);
+    }
+
+    private String substitutePortNumber(String document) {
+        return document.replaceAll("\\{\\{port\\}\\}", Integer.toString(port));
```

Comment:
Since we can't use `Library.getFileContent()` :-( we should at least be consistent and use the same replacement strings, i.e., `$port$`, `$sourceId$` and `$httpRoot$`.